File: news_server/modules/auth/presentation/eventbus/registry.ts
/**
 * EventBus Handler 注册表
 * 类似 Sjgz-Backend 的 interfaces/eventbus/server.go 中的 Registry
 */
import { AuthHandler } from './handler/auth'

export class Registry {
  Auth: AuthHandler

  constructor(authHandler: AuthHandler) {
    this.Auth = authHandler
  }
}

File: modules/crawl/interfaces/http/registry.go
package http

import (
	authconn "nfxnews/connections/auth"
	crawlapp "nfxnews/modules/crawl/application/crawl"
	"nfxnews/modules/crawl/interfaces/http/handler"
)

type Registry struct {
	App  *handler.CrawlHandler
	I18n *handler.I18nHandler
}

func NewRegistry(svc *crawlapp.Service, langs string, identity *authconn.Client) *Registry {
	return &Registry{App: handler.NewCrawlHandler(svc, identity), I18n: handler.NewI18nHandler(langs)}
}
